packages/headless/src/features/index.ts

```diff
@@ -196,18 +196,26 @@ export namespace RedirectionActions {
   export const buildPlanRequest = buildPlanRequestAlias;
 }
 
+export {
+  StateNeededByExecuteSearch,
+  ExecuteSearchThunkReturn,
+} from './search/search-actions';
+
 import {
-  StateNeededByExecuteSearch as StateNeededByExecuteSearchAlias,
-  ExecuteSearchThunkReturn as ExecuteSearchThunkReturnAlias,
-  executeSearch as executeSearchAlias,
   buildSearchRequest as buildSearchRequestAlias,
+  executeSearch as executeSearchAlias,
   fetchMoreResults as fetchMoreResultsAlias,
 } from './search/search-actions';
 export namespace SearchActions {
-  export type StateNeededByExecuteSearch = StateNeededByExecuteSearchAlias;
-  export type ExecuteSearchThunkReturn = ExecuteSearchThunkReturnAlias;
-  export const executeSearch = executeSearchAlias;
   export const buildSearchRequest = buildSearchRequestAlias;
+  /**
+   * Executes a search query.
+   * @param analyticsAction - The analytics action to log after a successful query.
+   */
+  export const executeSearch = executeSearchAlias;
+  /**
+   * Executes a search query and appends the results after the existing ones.
+   */
   export const fetchMoreResults = fetchMoreResultsAlias;
```

Comment:
When re-exporting types with `const`, all documentation attached to it is lost. I repeated the documentation here, but an approach to avoid repetition could be to use the `import export` statement.

However, the `import export` statement is blocked by `rollup-plugin-dts`, I can investigate replacing it.

Reply:
What if we create file exporting the actions we want to expose, and then here, instead of using namespaces, we would use a name alias. e.g.

```
// actions.ts
...
all actions
...

// public-actions.ts

export {action1, action2} from 'actions'

// index.ts

export * as SearchActions from 'public-actions'
```

Another alternative would be to just export actions/interfaces from an action file as we did originally, but this is hard to enforce, and could cause utilities to leak out. The middle `public-actions` file would help to avoid this.

Reply:
Unfortunately, I just tested and this doesn't seem to be possible with `rollup-plugin-dts` since it translates that to a namespaces containing consts just like we do, but doesn't copy the comments. Here's what it generates exactly:
```ts
/**
 * Executes a search query.
 * @param analyticsAction - The analytics action to log after a successful query.
 */
declare const executeSearch: AsyncThunkDefinition<'search/executeSearch', ExecuteSearchThunkReturn, SearchAction, AsyncThunkSearchOptions<StateNeededByExecuteSearch>>;
/**
 * Executes a search query and appends the results after the existing ones.
 */
declare const fetchMoreResults: AsyncThunkDefinition<'search/fetchMoreResults', ExecuteSearchThunkReturn, void, AsyncThunkSearchOptions<StateNeededByExecuteSearch>>;
declare const buildSearchRequest: (state: StateNeededByExecuteSearch) => SearchRequest;

type searchActions_d_StateNeededByExecuteSearch = StateNeededByExecuteSearch;
type searchActions_d_ExecuteSearchThunkReturn = ExecuteSearchThunkReturn;
declare const searchActions_d_executeSearch: typeof executeSearch;
declare const searchActions_d_fetchMoreResults: typeof fetchMoreResults;
declare const searchActions_d_buildSearchRequest: typeof buildSearchRequest;
declare namespace searchActions_d {
  export {
    searchActions_d_StateNeededByExecuteSearch as StateNeededByExecuteSearch,
    searchActions_d_ExecuteSearchThunkReturn as ExecuteSearchThunkReturn,
    searchActions_d_executeSearch as executeSearch,
    searchActions_d_fetchMoreResults as fetchMoreResults,
    searchActions_d_buildSearchRequest as buildSearchRequest,
  };
}
```